trusted-committer-article.md

```diff
@@ -0,0 +1,232 @@
+      
+## Reducing Barriers To Entry
+
+- Soliciting contributions is one of the things that is more challenging with
+  InnerSource compared to Open Source. There are two main reasons for that 
+    - The sheer number of potential contributors is lower in InnerSource
+    - Contributors will want to contribute during their work time. That means
+      they are more time constrained compared to doing Open Source after office
+      hours.
+- It is therefore super important to make the process for making contributions
+  and for onboarding contributors as frictionless as possible to avoid wasting
+  a contributors time. 
+- There are a number of things a TC can do to achieve that state
+    - have a good readme.md, providing detailed instructions on how to get,
+      build, test and use the SW developed by the InnerSource community.
+    - have a good contribute.md which outlines what is expected of the
+      contributor to make a contribution. It should answer common questions
+      like
+        - Who do I contact in case of questions and how to reach her?
+        - How do I submit a bug report or feature request?
+        - What is the definition of done for a contribution; code, tests, docs?
+        - What are conventions for code style, branching or committ messages, e. g.?
+        - What is expected of me in terms of supporting contributed code after
+          the contribution was accepted?
+        - What are the process steps that govern contributions?
+        - Is there a code of conduct or other guidelines to how the community
+          operates?
+- As a TC, I have to make sure there is someone in the community who is
+  available to answer questions of users and potential contributors.
+- Actively publishing a list of "Newbie Tasks" to actively solicit contributions,
+  rather than waiting for them to happen.
+- Also make it clear that non-code contributions are welcome, such as writing
+  documentation, organizing events or creating artwork.
+- Gig-Marketplace as a good example of how to do this.
+
+## Uplevelling Community Members
+
+- identifying talent
+- encouraging growth
+- mentoring community members (e. g. during PRs)
+
+## Advocating The Communities Needs
+
+- One major difference of InnerSource compared to Open Source software
+  development is the degree of freedom communities have. 
+- Naturally, an InnerSource community is more constrained because it has to fit
+  in the corporate context.
+- There are varying degrees of freedom, depending on whether your InnerSource
+  initiative focuses more on permissionless innovation or on shared development
+  of reusable components.
+- In any case, there can be times when the company interests are at odds with
+  those of the community. And while companies are more concerned with the
+  bottom line and thus the products produced by an InnerSource community,
+  InnerSource communities, on the other hand understand that a healthy
```

Comment:
This point might be made earlier in the Trusted committer section.  Also, I would say that this presupposes that healthy code is important for inner source which may or may not be true.

But really it is important for the TC to realize that the healthiness of the project is only relevant/important based on the number of contributors to the code.  Some code with no tests at all might be very very useful, and is so clear and obvious that more time spent wouldn't be worth while since only 2 maintainers and users are sharing the code.

Something about code health might be stated early on in the TC section at the top.

Reply:
I think we can state that TC effectiveness is going to be strongly related to code health. If the TC has to spend all their time validating and documenting workarounds for bugs, fragile architecture, or gaps between test and production environments they are not going to have as much time to help mentor contributors. This might be attributed to the process of InnerSource itself rather than the fragility of the code base. I want to give TCs (and their managers) the expectation that the contribution process may uncover problems but not be the cause of them.

Reply:
> Also, I would say that this presupposes that healthy code is important for inner source which may or may not be true.

I'd say: healthy code is important for the _long term_ health of an InnerSource community, but I think a healthy community can totally start with _less healthy_ code.

> But really it is important for the TC to realize that the healthiness of the project is only relevant/important based on the number of contributors to the code.

Not sure I get your point, @mrsanz . Care to elaborate?

@cewilliams : Excellent points!

Reply:
> I'd say: healthy code is important for the long term health of an InnerSource community, but I think a healthy community can totally start with less healthy code.

I think I'm attempting to make a semantic difference.  The difference is the difference between _long term health_ and _growth_.  All too often there is an assumption that a project is healthy or it isn't, when in reality it is a continuum. 

I agree with all of the points above from @cewilliams and @gruetter in this context.  Not as a static one time snapshot as was my initial impression of this section.  But rather in two parts.  Before the project exists and as it is growing.  

The second point I was trying to make is the case where projects don't get off the ground because they are too busy fixing cd/ci/documentation/testing - all the things related to code health.

Reply:
> I think we can state that TC effectiveness is going to be strongly related to code health. If the TC has to spend all their time validating and documenting workarounds for bugs, fragile architecture, or gaps between test and production environments they are not going to have as much time to help mentor contributors. This might be attributed to the process of InnerSource itself rather than the fragility of the code base. I want to give TCs (and their managers) the expectation that the contribution process may uncover problems but not be the cause of them.

This point could be mentioned here, but should also be part of the separate Contributor learning path video/writeups.

Reply:
> The second point I was trying to make is the case where projects don't get off the ground because they are too busy fixing cd/ci/documentation/testing - all the things related to code health.

Interesting. Never occurred to me that could be a problem, @mrsanz . I would summarize that as follows: _"There has to be a healthy balance between effort spent on code quality and growing the community.". Does that capture your thought, @mrsanz ?